projects/demo/src/modules/components/dialog/examples/9/pay-modal/pay-modal.component.ts

```diff
@@ -74,18 +74,22 @@ export class PayModal implements OnInit {
         saveCard: new FormControl(true),
     });
 
-    protected readonly context =
-        injectContext<TuiDialogContext<void, DataForPayCardModal>>();
-
     protected readonly iOS = inject(TUI_IS_IOS);
 
     protected cards: AccountCard[] = [];
     protected paymentMode: TuiValuesOf<typeof PaymentMode> = PaymentMode.ByNewCard;
     protected loading$ = new BehaviorSubject(false);
     protected payProcessing$ = new BehaviorSubject(false);
-    protected amount: number = this.context?.data?.amount ?? 0;
+    protected amount = 0;
     protected readonly PAYMENT_MODE = PaymentMode;
 
+    public readonly context =
+        injectContext<TuiDialogContext<void, DataForPayCardModal>>();
+
+    constructor() {
+        this.amount = this.context?.data?.amount ?? 0;
```

Comment:
Could you explain the reasons for these line changes, please  ?

Reply:
1. The field is changed from protected to public, as explained in the comment above.
2. Since public fields must be placed after protected ones, I changed their order.
3. After the change in order, the amount field cannot read the fields declared below it. Therefore, the only way to do it is in the constructor.

Reply:
Sounds like it's better to just use public for amount, it's a demo anyway. Looks/reads better than constructor.